arctic/store/metadata_store.py

```diff
@@ -0,0 +1,229 @@
+                if metadata == doc['metadata']:
+                    continue
+                doc['end_time'] = start_time
+                documents.append(doc)
+                doc = {'symbol': symbol, 'metadata': metadata, 'start_time': start_time}
+            else:
+                documents.append(doc)
+
+        self.insert_many(documents)
+
+    def write_metadata(self, symbol, metadata, start_time=None):
+        """
+        Update metadata entry for `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        start_time : `datetime.datetime`
+            when metadata becomes effective
+            Default: datetime.utcnow()
+        """
+        if start_time is None:
+            start_time = dt.utcnow()
+        old_metadata = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        if old_metadata is not None:
+            if old_metadata['start_time'] >= start_time:
+                raise ValueError('start_time={} is earlier than the last metadata @{}'.format(start_time,
+                                                                                              old_metadata['start_time']))
+            if old_metadata['metadata'] == metadata:
+                logger.warning('No change to metadata')
+                return metadata
+        elif metadata is None:
+            return
+
+        self.find_one_and_update({'symbol': symbol}, {'$set': {'end_time': start_time}},
+                                  sort=[('start_time', pymongo.DESCENDING)])
+        return self._write_metadata_entry(symbol, metadata, start_time)
+
+    def delete_last_metadata(self, symbol):
+        """
+        Delete current metadata of `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name to delete
+
+        Returns
+        -------
+        Deleted metadata
+        """
+        last_metadata = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        if last_metadata is None:
+            raise NoDataFoundException('No metadata found for symbol {}'.format(symbol))
+
+        self.find_one_and_delete({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        mongo_retry(self.find_one_and_update)({'symbol': symbol}, {'$unset': {'end_time': ''}},
+                                              sort=[('start_time', pymongo.DESCENDING)])
+
+        return last_metadata
+
+    def delete_all_metadata(self, symbol):
+        """
+        Delete all metadata of `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name to delete
+        """
+        logger.warning("Deleting entire metadata history for %r from %r" % (symbol, self._arctic_lib.get_name()))
+        self.delete_many({'symbol': symbol})
+        assert not self.has_symbol(symbol)
```

Comment:
I don't think this `assert` adds anything (and should be a real exception if this could go wrong).  I'd just remove this line.